packages/api/endpoints/granules.js

```diff
@@ -940,6 +949,159 @@ async function del(req, res) {
+  batchSize: z.number().positive().optional().default(100),
+  concurrency: z.number().positive().optional().default(100),
+  invalidBehavior: z.enum(['error', 'skip']).default('error'),
+  cmrGranuleUrlType: z.enum(['http', 's3', 'both']).default('both'),
+  s3MultipartChunkSizeMb: z.number().optional(),
+  executionName: z.string().optional(),
+  dbMaxPool: z.number().positive().optional().default(100),
+});
+const parsebulkChangeCollectionPayload = zodParser('bulkChangeCollection payload', bulkChangeCollectionSchema);
+
+/**
+ * Bulk move granules to a new collection.
+ *
+ * @param {object} req - The request object.
+ * @param {object} req.body - The request payload.
+ * @param {string} req.body.sourceCollectionId - The source collection ID.
+ * @param {string} req.body.targetCollectionId - The target collection ID.
+ * @param {number} [req.body.batchSize=100] - The batch size for processing granules.
+ * @param {number} [req.body.concurrency=100] - The per-file concurrency level for processing
+ * granules and granule records
+ * @param {string} [req.body.invalidBehavior='error'] - The behavior for invalid granules
+ * ('error' or 'skip').
+ * @param {number} [req.body.s3MultipartChunkSizeMb] - The S3 multipart chunk size in MB
+ * @param {string} [req.body.executionName] - Override to allow specifying an execution 'name'
+ * @param {object} req.testContext - The test context object
+ * @param {object} res - The response object.
+ * @returns {Promise<Object>} The response object with the execution ARN and message.
+ */
+async function bulkChangeCollection(req, res) {
+  const {
+    knex = await getKnexClient(),
+    sfnMethod = sfn,
+    workflow = 'MoveGranuleCollectionsWorkflow',
+  } = req.testContext || {};
+
+  const collectionPgModel = new CollectionPgModel();
+  const granulePgModel = new GranulePgModel();
+
+  if (!process.env.system_bucket) {
+    return res.boom.badRequest('API is misconfigured, system_bucket must be defined in the env variables');
+  }
+  if (!process.env.stackName) {
+    return res.boom.badRequest('API is misconfigured, stackName must be defined in the env variables');
+  }
+
+  const body = parsebulkChangeCollectionPayload(req.body);
+  if (isError(body)) {
+    return returnCustomValidationErrors(res, body);
+  }
+
+  //get collection
+  const pgCollection = await collectionPgModel.get(
+    knex,
+    deconstructCollectionId(body.sourceCollectionId)
+  );
+  const query = granulePgModel.queryBuilderSearch(knex, {
+    collection_cumulus_id: pgCollection.cumulus_id,
+  });
+  query.select('granule_id');
+  query.limit(body.batchSize);
+  const granules = await query;
+  if (granules.length === 0) {
+    return res.boom.notFound(
+      `No granules found for collection ${body.sourceCollectionId}`
+    );
+  }
+
+  const { name, version } = deconstructCollectionId(body.sourceCollectionId);
+  const executionName = body.executionName || uuidv4();
```

Comment:
We're defaulting to a uuid because we're really only using this in our S3 path and doesn't truly matter as long as it's unique?

Reply:
Correct - UUID is a good default (we use it for most other executions like this, the override is allowed in case users want to programmatically track this via their own convention for ease of AWS UI, etc)